inc/gpio.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+// GPIO HAL Interface
+
+#include <stdbool.h>
+#include <stdint.h>
+
+// IO Pin "address" to be used to change that pin's settings
+// Use bytes instead of words to save on memory also convenient for MSP430
```

Comment:
http://www.catb.org/esr/structure-packing/

This a pretty interesting page about struct padding.